Pspcl/Pspcl.Web/Controllers/ReportController.cs

```diff
@@ -67,5 +67,54 @@ public IActionResult StockOutReport()
 			}
 			return View();
 		}
+
+        [HttpGet]
+        public JsonResult FilteredStockInReport(DateTime? fromDate, DateTime? toDate)
+        {
+            try
+            {
+                var filteredstockInModels = _stockService.GetStockInModels();
+
+                if (fromDate.HasValue && toDate.HasValue)
+                {
+                    filteredstockInModels = filteredstockInModels.Where(s => s.Stock.GrnDate >= fromDate.Value && s.Stock.GrnDate <= toDate.Value)
+                        .OrderByDescending(s => s.Stock.GrnDate)
+                        .ToList();
+                }
+
+                return Json(filteredstockInModels);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Exception");
+            }
+            return Json("");
+
+        }
+
+        [HttpGet]
+        public JsonResult FilteredStockOutReport(DateTime? fromDate, DateTime? toDate)
+        {
+            try
+            {
+                var filteredstockOutModels = _stockService.GetStockOutModels();
+
+                if (fromDate.HasValue && toDate.HasValue)
+                {
+                    filteredstockOutModels = filteredstockOutModels.Where(s => s.CurrentDate >= fromDate.Value && s.CurrentDate <= toDate.Value)
```

Comment:
No null checks are there, if user select only one date, in that case, your LINQ query will throw exception

Reply:
User has to select both dates then click on filter button..If any date is null then it will show validation modal and also catch block will handle the exception.